frontend/src/lib/WidgetStateManager.ts

```diff
@@ -219,7 +218,14 @@ export class WidgetStateManager {
     // Create the button's triggerValue. Just like with a regular button,
     // `st.form_submit_button()` returns True during a rerun after
     // it's clicked.
-    this.createWidgetState(submitButton, { fromUi: true }).triggerValue = true
+    const submitButtons = this.formsData.submitButtons.get(formId)
+
+    // can have an empty list of submitButtons
+    if (submitButtons !== undefined && submitButtons.length > 0) {
+      // click the first submit button. We can choose any so we just choose first.
+      this.createWidgetState(submitButtons[0], { fromUi: true }).triggerValue =
```

Comment:
What happens if you have an app that uses multiple submit buttons in a form?

```python
import streamlit as st

with st.form("form"):
    if st.form_submit_button("submit button 1"):
        st.write("submit button 1 clicked")
    if st.form_submit_button("submit button 2"):
        st.write("submit button 2 clicked")
```

We currently support this (though it's surely rare). With this PR, it looks like clicking the second button will trigger the first button's logic? If so, I think that would be a regression we want to fix before merging.

Reply:
Oops thanks. I'll take a look at this later today when I have time!